File: base/src/main/java/me/shetj/base/http/manager/TokenManager.java
package me.shetj.base.http.manager;

import org.xutils.x;

import me.shetj.base.tools.file.SPUtils;
import me.shetj.base.tools.json.EmptyUtils;
import io.reactivex.Observable;
import me.shetj.base.tools.app.DataConfig;

/**
 * Created by admin on 2017/10/16.
 */

public class TokenManager {

	private static TokenManager instance = null;
	private TokenManager() {
	}

	public static TokenManager getInstance() {
		if (instance == null) {
			synchronized (TokenManager.class) {
				if (instance == null) {
					instance = new TokenManager();
				}
			}
		}
		return instance;
	}


	public Observable<String> getToken() {
		String cacheToken = TokenLoader.getInstance().getCacheToken();
		if (cacheToken.contains("token_fail")){
			return TokenLoader.getInstance().getNetTokenLocked();
		}else {
			return Observable.just(cacheToken);
		}
	}

	
	/**
	 * Is login boolean.
	 * @return the boolean
	 */
	public  boolean isLogin(){
		String token= (String) SPUtils.get(x.app().getApplicationContext(), DataConfig.PRE_CUSTOM_TOKEN,"");
		return EmptyUtils.isNotEmpty(token);
	}


	public void setToken(String token) {
		 SPUtils.put(x.app().getApplicationContext(), DataConfig.PRE_CUSTOM_TOKEN,token);
	}
}

File: testmybase/src/main/java/me/shetj/testmybase/common/TokenManager.kt
package shetj.me.base.common

import me.shetj.base.tools.file.SPUtils
import me.shetj.base.tools.json.EmptyUtils
import org.xutils.x
import shetj.me.base.configs.tag.SPKey.Companion.SAVE_TOKEN


/**
 *
 * @author shetj
 * @date 2017/10/16
 */

class TokenManager private constructor() {


    var token: String
        get() {
            val token = SPUtils[x.app().applicationContext, SAVE_TOKEN, ""] as String
            return if (EmptyUtils.isEmpty(token)) {
                ""
            } else token

        }
        set(token) = SPUtils.put(x.app().applicationContext, SAVE_TOKEN, token)
    val isLogin: Boolean
        get() {
            val token = SPUtils[x.app().applicationContext, SAVE_TOKEN, ""] as String
            return EmptyUtils.isNotEmpty(token)
        }

    companion object {


        private var instance: TokenManager? = null

        fun getTokenManager(): TokenManager? {
            if (instance == null) {
                synchronized(TokenManager::class.java) {
                    if (instance == null) {
                        instance = TokenManager()
                    }
                }
            }
            return instance
        }
    }


}
